clouddriver-web/src/test/groovy/com/netflix/spinnaker/clouddriver/controllers/CredentialsControllerSpec.groovy

```diff
@@ -58,6 +58,73 @@ class CredentialsControllerSpec extends Specification {
     parsedResponse == [[name: "test", environment: "env", accountType: "acctType", cloudProvider: "testProvider", type: "testProvider", requiredGroupMembership: ["test"], permissions: [READ:["test"], WRITE:["test"]], challengeDestructiveActions: false, primaryAccount: false]]
   }
 
+  /**
+   * Test to verify the use of the mandatory type (path) parameter,
+   * without passing the optional limit (query) parameter
+   */
+  void "credentials are listed by type"() {
+    setup:
+
+    def objectMapper = new ObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_NULL)
+    def credsRepo = new MapBackedAccountCredentialsRepository()
+    def accountDefRepo = new MapBackedAccountDefinitionRepository()
+    def credsProvider = new DefaultAccountCredentialsProvider(credsRepo)
+    def accountDefSrvc = Optional.of(new DefaultAccountDefinitionService(accountDefRepo))
+    accountDefRepo.save(new ManagedAccount("test1", "acctType1"))
+    accountDefRepo.save(new ManagedAccount("test2", "acctType1"))
+    accountDefRepo.save(new ManagedAccount("test3", "acctType2"))
+    accountDefRepo.save(new ManagedAccount("test4", "acctType2"))
+    def mvc = MockMvcBuilders.standaloneSetup(new CredentialsController(accountDefSrvc, new CredentialsConfiguration(), objectMapper, credsProvider)).build()
+
+    // path param:
+    def acctType = "acctType1"
+
+    when:
+    def result = mvc.perform(MockMvcRequestBuilders.get("/credentials/type/${acctType}").accept(MediaType.APPLICATION_JSON)).andReturn()
+
+    then:
+    result.response.status == 200
+
+    List<Map> parsedResponse = new JsonSlurper().parseText(result.response.contentAsString) as List
+
+    parsedResponse.every { acct -> acct.accountType == acctType }
+  }
+
+  /**
+   * Test to verify the use of the type (path)
+   * and limit (query) parameters
+   */
+  void "credentials are listed by type and with limit"() {
```

Comment:
Seems like we could use a test with startingAccountName too if I may ask for yet more work.

Reply:
Added.